commit ac393fa883f1365173bbc7cd4217bbd19661fa05
Author: Bernhard Rosenkranzer <bero@lindev.ch>
Date:   Wed Jan 18 13:38:27 2017 +0000

    Add support for processing lex source files with flex v 2.6.3
    
            PR 21059
    binutils* arlex.l: Support processing with flex 2.6.3.
            * deflex.l: Likewise.
    
    gas     * config/bfin-lex.l: Support processing with flex 2.6.3.
            * itbl-lex.l: Likewise.

diff --git a/binutils/arlex.l b/binutils/arlex.l
index bf186238c9..85f31fd707 100644
--- a/binutils/arlex.l
+++ b/binutils/arlex.l
@@ -1,4 +1,4 @@
-%option noinput nounput
+%option noinput nounput noyywrap
 
 %{
 /* arlex.l - Strange script language lexer */
@@ -88,7 +88,3 @@ int linenumber;
 "\n"	 	 { linenumber ++; return NEWLINE; }
 
 %%
-#ifndef yywrap
-/* Needed for lex, though not flex. */
-int yywrap(void) { return 1; }
-#endif
diff --git a/binutils/deflex.l b/binutils/deflex.l
index 0419f5f83b..2cbd9dd6d7 100644
--- a/binutils/deflex.l
+++ b/binutils/deflex.l
@@ -1,4 +1,4 @@
-%option noinput nounput
+%option noinput nounput noyywrap
 
 %{/* deflex.l - Lexer for .def files */
 
@@ -91,7 +91,3 @@ int linenumber;
 "@"	 	{ return '@';}
 ","		{ return ',';}
 %%
-#ifndef yywrap
-/* Needed for lex, though not flex. */
-int yywrap(void) { return 1; }
-#endif
diff --git a/gas/config/bfin-lex.l b/gas/config/bfin-lex.l
index 04fda4c7af..b2f0d84950 100644
--- a/gas/config/bfin-lex.l
+++ b/gas/config/bfin-lex.l
@@ -1,3 +1,5 @@
+%option noyywrap
+
 /* bfin-lex.l  ADI Blackfin lexer
    Copyright (C) 2005-2016 Free Software Foundation, Inc.
 
@@ -545,12 +547,3 @@ set_start_state (void)
 {
   BEGIN KEYWORD;
 }
-
-
-#ifndef yywrap
-int
-yywrap ()
-{
-  return 1;
-}
-#endif
diff --git a/gas/itbl-lex.l b/gas/itbl-lex.l
index a0b3653b5a..5339fe2938 100644
--- a/gas/itbl-lex.l
+++ b/gas/itbl-lex.l
@@ -1,3 +1,5 @@
+%option noyywrap
+
 /* itbl-lex.l
    Copyright (C) 1997-2016 Free Software Foundation, Inc.
 
@@ -102,11 +104,3 @@ HEX	[0-9A-Fa-f]
     return yytext[0];
   }
 %%
-
-#ifndef yywrap
-int
-yywrap ()
-  {
-    return 1;
-  }
-#endif
